app/controllers/tracks/concepts_controller.rb

```diff
@@ -1,34 +1,42 @@
 class Tracks::ConceptsController < ApplicationController
   before_action :use_track
+  before_action :use_concepts, only: :index
+  before_action :use_concept, only: %i[show start complete]
 
   allow_unauthenticated! :index, :show
 
   def authenticated_index
-    use_concepts
-
     if current_user.joined_track?(@track)
       render action: "index/joined"
     else
       render action: "index/unjoined"
     end
   end
 
-  def external_index
-    use_concepts
-  end
+  def external_index; end
 
   def authenticated_show
-    use_concept
-
     if current_user.joined_track?(@track)
+      @solution = Solution.for(current_user, @exercise)
+
       render action: "show/joined"
     else
       render action: "show/unjoined"
     end
   end
 
-  def external_show
-    use_concept
+  def external_show; end
+
+  def start
+    solution = Solution::Create.(@user_track, @exercise)
+    redirect_to edit_solution_path(solution.uuid)
+  end
+
+  def complete
+    ConceptExercise::Complete.(@exercise)
+    solution = Solution.for(current_user, @exercise)
+    solution.complete!
+    redirect_to action: :show
   end
```

Comment:
These are here for now, but I'll almost certainly move them at some stage to a different controller (ConceptExercise or ConceptSolutions or Solutions). But I want to work out some more paths first.